UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetGenerateFilesForProductIdentifierWithParameterexchangeSetStandardAsS57.cs

```diff
@@ -0,0 +1,143 @@
+                cleanUpBatchIdList.Add(batchId);
+                var downloadFolderPath = await FileContentHelper.CreateExchangeSetFile(apiResponse, FssJwtToken);
+                downloadedFolderPathList.Add(downloadFolderPath);
+            }
+        }
+
+         //PBI 143370: Change related to additional param (From Boolean to String)
+        [Test]
+        [Category("QCOnlyTest-AIODisabled")]
+        public async Task WhenICallProductIdentifiersApiWithS57ForParameterexchangeSetStandard_ThenAProductTxtFileIsGenerated()
+        {
+            foreach (var downloadedFolderPath in downloadedFolderPathList)
+            {
+                var checkFile = FssBatchHelper.CheckforFileExist(Path.Combine(downloadedFolderPath, Config.ExchangeSetProductFilePath), Config.ExchangeSetProductFile);
+                Assert.IsTrue(checkFile, $"File not Exist in the specified folder path : {Path.Combine(downloadedFolderPath, Config.ExchangeSetProductFilePath)}");
+
+                //Verify Product.txt file content
+                var apiScsResponse = await ScsApiClient.GetScsCatalogueAsync(Config.ExchangeSetProductType, Config.ExchangeSetCatalogueType, ScsJwtToken);
+                Assert.AreEqual(200, (int)apiScsResponse.StatusCode, $"Incorrect status code is returned {apiScsResponse.StatusCode}, instead of the expected status 200.");
+                var apiResponseDetails = await apiScsResponse.ReadAsStringAsync();
+                dynamic apiScsResponseData = JsonConvert.DeserializeObject(apiResponseDetails);
+
+                FileContentHelper.CheckProductFileContent(Path.Combine(downloadedFolderPath, Config.ExchangeSetProductFilePath, Config.ExchangeSetProductFile), apiScsResponseData);
+            }
+        }
+
+         //PBI 143370: Change related to additional param (From Boolean to String)
+        [Test]
+        [Category("QCOnlyTest-AIODisabled")]
+        public void WhenICallProductIdentifiersApiWithS57ForParameterexchangeSetStandard_ThenAReadMeTxtFileIsGenerated()
+        {
+            foreach (var downloadedFolderPath in downloadedFolderPathList)
+            {
+                var checkFile = FssBatchHelper.CheckforFileExist(Path.Combine(downloadedFolderPath, Config.ExchangeSetEncRootFolder), Config.ExchangeReadMeFile);
+                Assert.IsTrue(checkFile, $"{Config.ExchangeReadMeFile} File not Exist in the specified folder path : {Path.Combine(downloadedFolderPath, Config.ExchangeSetEncRootFolder)}");
+
+                //Verify README.TXT file content
+                FileContentHelper.CheckReadMeTxtFileContent(Path.Combine(downloadedFolderPath, Config.ExchangeSetEncRootFolder, Config.ExchangeReadMeFile));
+            }
+        }
+
+         //PBI 143370: Change related to additional param (From Boolean to String)
+        [Test]
+        [Category("QCOnlyTest-AIODisabled")]
+        public async Task WhenICallProductIdentifiersApiWithS57ForParameterexchangeSetStandard_ThenACatalogFileIsGenerated()
```

Comment:
Can we change the method name as below & Class name should have s in it ProductIdentifier => ProductIdentifiers
```suggestion
        public async Task WhenICallProductIdentifiersApiWithS57ExchangeSetStandardParameter_ThenACatalogFileIsGenerated()
```
**_Note:_** Please check other classes and test methods same as this and do the required changes.

Reply:
Resolved.Thanks